admin/admin-links.php

```diff
@@ -157,49 +157,90 @@ protected function new_translation_link( string $link, PLL_Language $language ):
 	/**
 	 * Returns the html markup for a post translation link.
 	 *
-	 * @since 1.4
+	 * @since 3.8
 	 *
-	 * @param int $post_id The translation post id.
+	 * @param WP_Post $post The post.
+	 * @param string  $mode Optional. How the link should be displayed: with a pen icon or a language's flag.
+	 *                      Possible values are `metabox`, `icon`, and `flag`. Default is `metabox`.
 	 * @return string
+	 *
+	 * @phpstan-param 'metabox'|'icon'|'flag' $mode
 	 */
-	public function edit_post_translation_link( int $post_id ): string {
-		$language = $this->model->post->get_language( $post_id );
+	public function get_edit_post_link_html( WP_Post $post, string $mode = 'metabox' ): string {
+		$language = $this->model->post->get_language( $post->ID );
 
 		if ( empty( $language ) ) {
 			// Should not happen.
 			return '';
 		}
 
-		$link = (string) get_edit_post_link( $post_id );
-		return $this->edit_translation_link( $link, $language );
+		$url = (string) get_edit_post_link( $post->ID );
+		return $this->edit_translation_link( $url, $language, $mode, $post->ID, $post->post_title );
 	}
 
 	/**
 	 * Returns the html markup for a translation link.
 	 *
-	 * @since 2.6
-	 * @since 3.8 Visibility changed from `public` to `protected`.
+	 * @since 3.8
 	 *
-	 * @param string       $link     The translation link.
-	 * @param PLL_Language $language The language of the translation.
+	 * @param string       $url       URL of the edition link.
+	 * @param PLL_Language $language  Language of the item.
+	 * @param string       $mode      How the link should be displayed: with a pen icon or a language's flag.
+	 *                                Possible values are `metabox`, `icon`, and `flag`.
+	 * @param int          $item_id   ID of the item. Used only in `icon` mode.
+	 * @param string       $item_name Name of the item. Not used in `metabox` mode.
 	 * @return string
+	 *
+	 * @phpstan-param 'metabox'|'icon'|'flag' $mode
 	 */
-	protected function edit_translation_link( string $link, PLL_Language $language ): string {
-		if ( empty( $link ) ) {
-			return sprintf(
-				'<span title="%s" class="pll_icon_edit wp-ui-text-icon"></span>',
+	private function edit_translation_link( string $url, PLL_Language $language, string $mode, int $item_id, string $item_name ): string {
```

Comment:
I suggest to rename to `get_edit_item_link_html()`.
I would move `$mode` at the beginning or the end because it's very different from the 4 other params which are all related to the item.
I'd like other names for modes.